File: gr/sidereal-time.fs
\ sidereal time (jmcc) #9

: p 15 ;
: z p 0 do 100 6000 ExpRand.ir 1 2 6 Rand.ir loop p 3 * mce ;
: f 40 300 ExpRand.ir 40 300 ExpRand.ir 12 0 XLine.kr ;
: t f 0 0.1 0.9 Rand.ir LFPulse.ar 0.002 * 0 0 8 Rand.ir LFNoise2.kr Max * ;
: rev unmce swap 2 mce ;
: cmb { z } z 0.6 0.1 0.6 Rand.ir 8 CombN z rev + ;

' cmb texture-post-proc

t 1 0 1 z Klank 2 clone Distort 0.1 *
4 4 6 inf overlap-texture

File: gr/sidereal-time.hs
-- sidereal time (jmcc) #9

import Sound.SC3 {- hsc3 -}
import Sound.SC3.Lang.Control.OverlapTexture {- hsc3-lang -}

sidereal_time :: UGen
sidereal_time =
  let p = 15
      z = let y = let fr = uclone' 'α' p (expRand 'β' 100 6000)
                      rt = uclone' 'γ' p (rand 'δ' 2 6)
                  in klankSpec fr (replicate p 1) rt
          in uclone 'ε' 2 y
      f = xLine KR (expRand 'ζ' 40 300) (expRand 'η' 40 300) 12 DoNothing
      t = let e = lfNoise2 'θ' KR (rand 'ι' 0 8)
          in lfPulse AR f 0 (rand 'κ' 0.1 0.9) * 0.002 * max 0 e
      o = distort (klank t 1 0 1 (mceTranspose z)) * 0.1
  in combN o 0.6 (rand 'λ' 0.1 0.6) 8 + mceReverse o

main :: IO ()
main = overlapTextureU (4,4,6,maxBound) sidereal_time

{-
audition (out 0 sidereal_time)
-}
